enclave_wrangler/main.py

```diff
@@ -242,27 +270,75 @@ def run(input_csv_folder_path):
         # and if needed we can also save the json data in upd_cs_ver_expression_items_dict
         # premade_codeset_id is stored in the codeset_id column in the csv files, save the id in the enclave_codeset_id column
         # update when it was uploaded as well, Stephane 3/15/22
-        code_sets_df.set_index('codeset_id', inplace=True)
+        try:
+            code_sets_df.set_index('codeset_id', inplace=True)
+        except Exception as e:
+            print(e)
+
         code_sets_df.at[premade_codeset_id, 'enclave_codeset_id'] = codeset_id
         code_sets_df.at[premade_codeset_id, 'enclave_codeset_id_updated_at'] = _datetime_palantir_format()
-        code_sets_df.reset_index()
+        code_sets_df = code_sets_df.reset_index()
         # return response_json
 
 
     # write out the update csv file with the enclave_codeset_id
     # print('before terminating write out the updated code_sets.csv file here')
     # date_str = datetime.now().strftime('%Y_%m_%d_%H_%M')
     # output_filename = 'code_sets_updated_' + date_str + '.csv'
+    # TODO: fix creates these columns depending on how we're doing indexing: Unnamed: 0	Unnamed: 0.1, etc
     output_filename = 'code_sets.csv'
     code_sets_df.to_csv(os.path.join(input_csv_folder_path, output_filename), index=True, encoding='utf-8')
 
-    # Save to persistence layer
+    return code_sets_df
+
+
+def left_join_update(df, df2):
+    """Does a left join, but where column names are the same, keeps right value if exists, else left value."""
+    new_df = df.merge(
+        df2, how='left', left_on='internal_id', right_on='codeset_id')
+
+    xy_col_counts: Dict[str, int] = {}
+    for col in list(new_df.columns):
+        if any([col.endswith(x) for x in ['_x', '_y']]):
+            common_col = col[:-2]
+            if common_col not in xy_col_counts:
+                xy_col_counts[common_col] = 0
+            xy_col_counts[common_col] += 1
+
+    xy_cols: List[str] = [k for k, v in xy_col_counts.items() if v == 2]
+    for col in xy_cols:
+        new_df[col] = new_df[col + '_y'].fillna(new_df[col + '_x'])
+        new_df = new_df.drop([col + '_x', col + '_y'], axis=1)
+
+    return new_df
+
+
+def persist_to_db(code_sets_df) -> pd.DataFrame:
+    """Save updated items to persistence layer"""
+    # Vars
+    join_col = 'codeset_id'
+    update_cols = ['enclave_codeset_id', 'enclave_codeset_id_updated_at', 'concept_set_name']
+
+    # Read
     persistence_csv_path = os.path.join(PROJECT_ROOT, 'data', 'cset.csv')
-    code_sets_df_limited = code_sets_df[['enclave_codeset_id', 'enclave_codeset_id_updated_at', 'concept_set_name']]
-    persistence_df = pd.read_csv(persistence_csv_path).fillna('')
-    persistence_df_new = persistence_df.merge(
-        code_sets_df_limited, how='left', left_on='internal_id', right_on='codeset_id').fillna('')
+    persistence_df = _load_standardized_input_df(persistence_csv_path)
+
+    # Subset
+    try:
```

Comment:
@Sigfried @stephanieshong Siggie added this code. This allows us to handle cases where `codeset_id` is either an index or just a normal column. I refactored it slightly. to try and make it clearer what's going on. I don't mind leaving this here, because I'm not entirely sure how I feel about the indexing right now, but I'm fine with either way.